staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_common.go

```diff
@@ -226,9 +235,13 @@ func (c *controllerCommon) GetDiskLun(diskName, diskURI string, nodeName types.N
 		if disk.Lun != nil && (disk.Name != nil && diskName != "" && strings.EqualFold(*disk.Name, diskName)) ||
 			(disk.Vhd != nil && disk.Vhd.URI != nil && diskURI != "" && strings.EqualFold(*disk.Vhd.URI, diskURI)) ||
 			(disk.ManagedDisk != nil && strings.EqualFold(*disk.ManagedDisk.ID, diskURI)) {
-			// found the disk
-			klog.V(2).Infof("azureDisk - find disk: lun %d name %q uri %q", *disk.Lun, diskName, diskURI)
-			return *disk.Lun, nil
+			if disk.ToBeDetached != nil && *disk.ToBeDetached {
```

Comment:
would `ToBeDetached` end in an error in 1.15 since it's got the older sdk?

Reply:
that's ok, when I do the cherry-pick,  I will remove this code change since 1.15 don't have `ToBeDetached` field.

Reply:
@andyzhangx - we were doing some testing with https://github.com/kubernetes/kubernetes/pull/84958. Looks like with this we don't need those changes ?

Reply:
@kkmsft I think that's ok, it's a double check, if there is `ToBeDetached` flag, then we should respect that. and if #84958 fixed that, we never go to that line of code.